### PR TITLE
Changes on 404, robots, and sitemap

### PR DESCRIPTION
**Pull request by stsewd**
- Drop check for project privacy level, follow the privacy level from the version instead.
- Only serve robots.txt if the version is public
- Only serve the sitemap if there is at least one public version
- Serve the custom 404 page from a version if the current user  has permissions over that version to allow leaking anything.

These changes aren't ported to the serve app, only on proxito. There is a PR on .com adapting the code to this new changes.

Ref https://github.com/readthedocs/readthedocs.org/pull/6194/

### CODE REVIEW

File: readthedocs/proxito/views/serve.py [hunk 1]
```diff
@@ -323,7 +322,7 @@ def get(self, request, project):
             include_file=False,
             version_type=self.version_type,
         )
-        path = os.path.join(storage_path, 'robots.txt')
+        path = f'{storage_path}/robots.txt'
```
Comment (stsewd) [inline readthedocs/proxito/views/serve.py]: We shouldn't use path.join for storage paths, there are more in this file

**Comment by humitos** [comment]
> These changes aren't ported to the serve app, only on proxito. There is a PR on .com adapting the code to this new changes.

Should we wait until we are fully on El Proxito to merge this? (we are really close to that)

**Comment by stsewd** [comment]
> Should we wait until we are fully on El Proxito to merge this? (we are really close to that)

Don't think is necessary, I mean, there isn't a big difference from what we have and these changes

File: readthedocs/proxito/views/serve.py [hunk 2]
```diff
@@ -266,14 +264,17 @@ def get(self, request, proxito_path, template_name='404.html'):
         if default_version_slug != version_slug:
             versions.append(default_version_slug)
         for version_slug_404 in versions:
+            if not self.allowed_user(request, final_project, version_slug_404):
```
Comment (ericholscher) [inline readthedocs/proxito/views/serve.py]: Hmm. This is going to add even more queries to our slow 404's. I'm a bit worried about that. I guess it won't cause a query on the .org, which is where i'm most worried about it, so it should be fine.

Comment (stsewd) [reply readthedocs/proxito/views/serve.py]: Yeah, on .org that function just returns True, it doesn't query the db.

**Comment by stsewd** [comment]
This PR is needed in .com to math this changes https://github.com/readthedocs/readthedocs-corporate/pull/875

**Pull request merged**